src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -25,20 +26,449 @@ using neural_net::model_backend;
+    data_iterator::parameters iterator_params) const {
+    return std::unique_ptr<data_iterator>(
+        new simple_data_iterator(iterator_params));
+}
+
+std::unique_ptr<data_iterator> drawing_classifier::create_iterator(
+    gl_sframe data, bool is_train,
+    std::vector<std::string> class_labels) const {
+  data_iterator::parameters data_params;
+  data_params.data = std::move(data);
+
+  if (!is_train) {
+    data_params.class_labels = std::move(class_labels);
+  }
+
+  data_params.is_train = is_train;
+  data_params.target_column_name = read_state<flex_string>("target");
+  data_params.feature_column_name = read_state<flex_string>("feature");
+  return create_iterator(data_params);
+}
+
+void drawing_classifier::init_train(gl_sframe data,
+                                    std::string target_column_name,
+                                    std::string feature_column_name,
+                                    variant_type validation_data,
+                                    std::map<std::string, flexible_type> opts) {
+
+  // Read user-specified options.
+  init_options(opts);
+
+  // Choose a random seed if not set.
+  if (read_state<flexible_type>("random_seed") == FLEX_UNDEFINED) {
+    std::random_device random_device;
+    int random_seed = static_cast<int>(random_device());
+    add_or_update_state({{"random_seed", random_seed}});
+  }
+
+  // Perform validation split if necessary.
+  std::tie(training_data_, validation_data_) = init_data(data, validation_data);
+
+  // Begin printing progress.
+  // TODO: Make progress printing optional.
+  init_table_printer(!validation_data_.empty());
+
+  add_or_update_state({{"target", target_column_name},
+                       {"feature", feature_column_name}});
+
+  // Bind the data to a data iterator.
+  training_data_iterator_ =
+      create_iterator(training_data_,
+                      /* is_train */ true, /* class labels */ {});
+
+  const std::vector<std::string>& classes =
+      training_data_iterator_->class_labels();
+  add_or_update_state({{"classes", flex_list(classes.begin(), classes.end())}});
+
+  // Bind the validation data to a data iterator.
+  if (!validation_data_.empty()) {
+    validation_data_iterator_ =
+        create_iterator(validation_data_,
+                        /* is_train */ false, /* class labels */ classes);
+  } else {
+    validation_data_iterator_ = nullptr;
+  }
+
+  // Instantiate the compute context.
+  training_compute_context_ = create_compute_context();
+  if (training_compute_context_ == nullptr) {
+    log_and_throw("No neural network compute context provided");
+  }
+
+  // Report to the user what GPU(s) is being used.
+  std::vector<std::string> gpu_names = training_compute_context_->gpu_names();
+  if (gpu_names.empty()) {
+    logprogress_stream << "Using CPU to create model";
+  } else {
+    std::string gpu_names_string = gpu_names[0];
+    for (size_t i = 1; i < gpu_names.size(); ++i) {
```

Comment:
Can you use the utility function now that it is in master? https://github.com/apple/turicreate/pull/2380

Reply:
Done.